### ISSUE TITLE
couldn't reproduce the result on IEMOCAP

### ISSUE THREAD

**Issue by jellyfishovo** [zerohd4869/MM-DFN#7]
Greetings!
I run the model on IEMOCAP dataset ,however i can not get the result as the paper gives
some parameters are as follows(basically i dont change them)
Namespace(no_cuda=False, dataset='IEMOCAP', data_dir='../data/iemocap/IEMOCAP_features.pkl', multi_modal=True, modals='avl', mm_fusion_mthd='concat_subsequently', use_modal=True, base_model='LSTM', graph_model=True, graph_type='GDF', graph_construct='direct', use_gcn=False, nodal_attention=True, use_topic=True, use_residue=True, av_using_lstm=False, active_listener=False, attention='general', use_crn_speaker=True, speaker_weights='3-0-1', use_speaker=False, reason_flag=False, epochs=30, batch_size=32, valid_rate=0.0, modal_weight=1.0, Deep_GCN_nlayers=16, lr=0.0003, l2=0.0001, rec_dropout=0.1, dropout=0.4, alpha=0.2, lamda=0.5, gamma=0.5, windowp=10, windowf=10, multiheads=6, loss='FocalLoss', class_weight=False, save_model_dir='../outputs/iemocap_demo/', tensorboard=False, test_label=False, load_model='../outputs/iemocap_demo/model_4.pkl', seed=2021)

i wonder what is the detailed parameter  to get the aforementioned result
thanks!!!!

**Comment by zerohd4869** [comment]
> Greetings! I run the model on IEMOCAP dataset ,however i can not get the result as the paper gives some parameters are as follows(basically i dont change them) Namespace(no_cuda=False, dataset='IEMOCAP', data_dir='../data/iemocap/IEMOCAP_features.pkl', multi_modal=True, modals='avl', mm_fusion_mthd='concat_subsequently', use_modal=True, base_model='LSTM', graph_model=True, graph_type='GDF', graph_construct='direct', use_gcn=False, nodal_attention=True, use_topic=True, use_residue=True, av_using_lstm=False, active_listener=False, attention='general', use_crn_speaker=True, speaker_weights='3-0-1', use_speaker=False, reason_flag=False, epochs=30, batch_size=32, valid_rate=0.0, modal_weight=1.0, Deep_GCN_nlayers=16, lr=0.0003, l2=0.0001, rec_dropout=0.1, dropout=0.4, alpha=0.2, lamda=0.5, gamma=0.5, windowp=10, windowf=10, multiheads=6, loss='FocalLoss', class_weight=False, save_model_dir='../outputs/iemocap_demo/', tensorboard=False, test_label=False, load_model='../outputs/iemocap_demo/model_4.pkl', seed=2021)
> 
> i wonder what is the detailed parameter to get the aforementioned result thanks!!!!

Hi. From your parameter information, it appears that you did not use our model ( `--reason_flag` should be set to True. `--class_weight` should be True for IEMOCAP).  Other optimal hyper-parameters differ under different experimental configurations, such as the version of CUDA and PyTorch. 

 I suggest running the `run_train_ie.sh` file directly to reproduce our work, as it contains the candidate hyperparameter search ranges.

Please feel free to ask me if you have any further questions after running the sh script to search for optimal parameters.

**Comment by jellyfishovo** [comment]
> > Greetings! I run the model on IEMOCAP dataset ,however i can not get the result as the paper gives some parameters are as follows(basically i dont change them) Namespace(no_cuda=False, dataset='IEMOCAP', data_dir='../data/iemocap/IEMOCAP_features.pkl', multi_modal=True, modals='avl', mm_fusion_mthd='concat_subsequently', use_modal=True, base_model='LSTM', graph_model=True, graph_type='GDF', graph_construct='direct', use_gcn=False, nodal_attention=True, use_topic=True, use_residue=True, av_using_lstm=False, active_listener=False, attention='general', use_crn_speaker=True, speaker_weights='3-0-1', use_speaker=False, reason_flag=False, epochs=30, batch_size=32, valid_rate=0.0, modal_weight=1.0, Deep_GCN_nlayers=16, lr=0.0003, l2=0.0001, rec_dropout=0.1, dropout=0.4, alpha=0.2, lamda=0.5, gamma=0.5, windowp=10, windowf=10, multiheads=6, loss='FocalLoss', class_weight=False, save_model_dir='../outputs/iemocap_demo/', tensorboard=False, test_label=False, load_model='../outputs/iemocap_demo/model_4.pkl', seed=2021)
> > i wonder what is the detailed parameter to get the aforementioned result thanks!!!!
> 
> Hi. From your parameter information, it appears that you did not use our model ( `--reason_flag` should be set to True. `--class_weight` should be True for IEMOCAP). Other optimal hyper-parameters differ under different experimental configurations, such as the version of CUDA and PyTorch.
> 
> I suggest running the `run_train_ie.sh` file directly to reproduce our work, as it contains the candidate hyperparameter search ranges.
> 
> Please feel free to ask me if you have any further questions after running the sh script to search for optimal parameters.
Thanks, it works!!!!

**Comment by jellyfishovo** [comment]
> > Greetings! I run the model on IEMOCAP dataset ,however i can not get the result as the paper gives some parameters are as follows(basically i dont change them) Namespace(no_cuda=False, dataset='IEMOCAP', data_dir='../data/iemocap/IEMOCAP_features.pkl', multi_modal=True, modals='avl', mm_fusion_mthd='concat_subsequently', use_modal=True, base_model='LSTM', graph_model=True, graph_type='GDF', graph_construct='direct', use_gcn=False, nodal_attention=True, use_topic=True, use_residue=True, av_using_lstm=False, active_listener=False, attention='general', use_crn_speaker=True, speaker_weights='3-0-1', use_speaker=False, reason_flag=False, epochs=30, batch_size=32, valid_rate=0.0, modal_weight=1.0, Deep_GCN_nlayers=16, lr=0.0003, l2=0.0001, rec_dropout=0.1, dropout=0.4, alpha=0.2, lamda=0.5, gamma=0.5, windowp=10, windowf=10, multiheads=6, loss='FocalLoss', class_weight=False, save_model_dir='../outputs/iemocap_demo/', tensorboard=False, test_label=False, load_model='../outputs/iemocap_demo/model_4.pkl', seed=2021)
> > i wonder what is the detailed parameter to get the aforementioned result thanks!!!!
> 
> Hi. From your parameter information, it appears that you did not use our model ( `--reason_flag` should be set to True. `--class_weight` should be True for IEMOCAP). Other optimal hyper-parameters differ under different experimental configurations, such as the version of CUDA and PyTorch.
> 
> I suggest running the `run_train_ie.sh` file directly to reproduce our work, as it contains the candidate hyperparameter search ranges.
> 
> Please feel free to ask me if you have any further questions after running the sh script to search for optimal parameters.

Thanks, it works!!!

**Comment by jellyfishovo** [comment]
Sorry to bother you again
While i have trouble in comprehending the code, the paper illustrates that the input features are consists of context embedding and speaker embedding,but i cant find the detailed code of this part, so i wonder whether the dataset files (.pkl) have been processed by concatnating this two embeddings instead of just some simple binarization processes...
Thanks again !!!

**Comment by zerohd4869** [comment]
> Sorry to bother you again While i have trouble in comprehending the code, the paper illustrates that the input features are consists of context embedding and speaker embedding,but i cant find the detailed code of this part, so i wonder whether the dataset files (.pkl) have been processed by concatnating this two embeddings instead of just some simple binarization processes... Thanks again !!!

The code implementation for equations 1-3 in the paper can be found in the `model.py` file. Specifically, audio feature processing can be found in lines 1065-1090, video feature processing can be found in lines 1094-1121, and text feature processing can be found in lines 1132-1154.
